lib/validate.js

```diff
@@ -1,60 +1,89 @@
+const EventEmitter = require('events')
 const tv4 = require('tv4')
 const fetch = require('node-fetch')
 const {Table} = require('tableschema')
 
-const {File, isUrl} = require('data.js')
-
-async function validate(datasetObj) {
-  try {
-    await validateMetadata(datasetObj.descriptor)
-    for (let i = 0; i < datasetObj.resources.length; i++) {
-      // Run data validation against schema if file format is either csv or tsv:
-      if (['csv', 'tsv'].includes(datasetObj.resources[i].descriptor.format)) {
-        await validateData(
-          datasetObj.resources[i].descriptor,
-          datasetObj.resources[i].path
-        )
+const {Dataset, File, isUrl} = require('data.js')
+
+class Validator extends EventEmitter {
+  constructor({identifier}={}) {
+    super()
+    this.identifier = identifier
+  }
+
+
+  async validate() {
+    this.emit('message', 'Loading the Dataset object...')
+    const datasetObj = await Dataset.load(this.identifier)
+    try {
+      this.emit('message', {name: 'datapackage.json', status: 'validating'})
+      await this.validateMetadata(datasetObj.descriptor)
+      this.emit('message', {name: 'datapackage.json', status: 'done'})
+      for (let i = 0; i < datasetObj.resources.length; i++) {
```

Comment:
Trivial:
I would suggest to use 
```
for (resource of datasetObj.resources)
``` 
from ES6, which looks better, and then you don't need to get resource by index everywhere, e.g. -
- `resource.descriptor.name`
VS
- `datasetObj.resources[i].descriptor.name`

Reply:
@AcckiyGerman agree... the ES6 looks much better - will do a commit!